dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt

```diff
@@ -0,0 +1,287 @@
+            searchResponseForName.contains(expectedCompany),
+            "The search results do not contain the expected company.",
+        )
+        assertTrue(
+            searchResponseForIdentifier.contains(expectedCompany),
+            "The search results do not contain the expected company.",
+        )
+    }
+
+    @Test
+    fun `search for name and check the ordering of results`() {
+        val testString = "unique-test-string-${UUID.randomUUID()}"
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val companyList = createCompaniesForTestingOrdering(testString)
+        for (company in companyList) {
+            val uploadedCompany = apiAccessor.companyDataControllerApi.postCompany(company)
+            uploadTestDataSet(uploadedCompany.companyId)
+        }
+        val sortedCompanyNames = apiAccessor.companyDataControllerApi.getCompaniesBySearchString(
+            searchString = testString,
+        ).map { it.companyName }
+        assertEquals(
+            listOf(testString, company2, "${testString}2", company5, "3$testString", company9),
+            sortedCompanyNames.filter { it != company8 && it != company3 },
+        )
+        assertEquals(
+            listOf(company3, company2, "${testString}2", company5, company8, company9),
+            sortedCompanyNames.filter { it != "3$testString" && it != testString },
+        )
+
+        val otherCompanyNames = apiAccessor.companyDataControllerApi.getCompaniesBySearchString(
+            searchString = "other_name",
+        ).map { it.companyName }
+        assertTrue(otherCompanyNames.contains(company8))
+        assertFalse(otherCompanyNames.contains("Company 7"))
+    }
+
+    private fun createCompaniesForTestingOrdering(inputString: String): List<CompanyInformation> {
+        return listOf(
+            CompanyInformation(
+                company9, "",
+                listOf(
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.isin, "3$inputString"),
+                ),
+                "", listOf(),
+            ),
+            CompanyInformation(company8, "", listOf(), "", listOf("3$inputString", "other_name")),
+            CompanyInformation("3$inputString", "", listOf(), "", listOf()),
+            CompanyInformation(company5, "", listOf(), "", listOf("${inputString}2")),
+            CompanyInformation("${inputString}2", "", listOf(), "", listOf()),
+            CompanyInformation(
+                company3, "",
+                listOf(
+                    apiAccessor.createCompanyIdentifier(CompanyIdentifier.IdentifierType.isin, inputString),
+                ),
+                "", listOf(),
+            ),
+            CompanyInformation(company2, "", listOf(), "", listOf(inputString)),
+            CompanyInformation(inputString, "", listOf(), "", listOf()),
+        )
+    }
+    private fun uploadTestDataSet(companyId: String): DataMetaInformation {
```

Comment:
This uploads specificly a financials dataset. So it should not be named as generic as now.

Reply:
Oder checkt mal ob es nicht bereits eine util Methode gibt die das macht.  Erscheint mir bereits in Vergangenheit gebraucht worden zu sein.

Reply:
I find it can be named generic if the data type does not matter. There is no ApiAccessor method for doingthe same thing.

Reply:
I think we can add a "fincials" to the name of the method, as it always adds a financial dataset. Would be more in line with what the method does.

Reply:
Since the used data and uploader Function is hard-coded to a euTaxonomyFinancials dataset I do agree with @emanueldfine .